AK/StringUtils.cpp

```diff
@@ -351,6 +351,23 @@ Optional<size_t> find_last(StringView const& haystack, char needle)
     return {};
 }
 
+Optional<size_t> find_last(StringView const& haystack, StringView const& needle)
+{
+    if (needle.length() > haystack.length()) {
+        return {};
+    } else if (needle == haystack) {
+        return 0;
+    }
+
+    for (auto i = haystack.length() - needle.length() + 1; i > 0; --i) {
```

Comment:
You start your loop from `i = haystack.length() - needle.length() + 1`, but then only ever use `i - 1` inside the loop. Is there a need to offset `i` by 1 that I'm not seeing?

Reply:
I think this is required unless you notice a different way. The `- 1` is needed `i` is unsigned. The loop will stop when the index is `0` and never be able to match a string with starts at the beginning.

Because the `- 1`, the `+ 1` is needed so matches can be made at the end of the string.

These boundary conditions are tested in the unit tests. If you can find a better way, I am open to it.

Reply:
Ah I see, makes sense. I would have just done

```cpp
for (int i = haystack.length() - needle.length(); i >= 0; --i) {
    // Use `i`
```

But I guess a narrowing conversation to `int` isn't ideal now that you point it out